src/cmd/services/m3aggregator/main/main.go

```diff
@@ -83,19 +86,46 @@ func main() {
 		SetMetricsSamplingRate(cfg.Metrics.SampleRate()).
 		SetReportInterval(cfg.Metrics.ReportInterval())
 
-	// Create the raw TCP server options.
-	rawTCPAddr := cfg.RawTCP.ListenAddress
-	rawTCPServerScope := scope.SubScope("rawtcp-server").Tagged(map[string]string{"server": "rawtcp"})
-	iOpts := instrumentOpts.SetMetricsScope(rawTCPServerScope)
-	rawTCPServerOpts := cfg.RawTCP.NewServerOptions(iOpts)
+	var (
+		m3msgAddr        string
+		m3msgServerOpts  m3msg.Options
+		rawTCPAddr       string
+		rawTCPServerOpts rawtcp.Options
+		httpAddr         string
+		httpServerOpts   http.Options
+	)
+	if cfg.M3Msg != nil {
+		// Create the M3Msg server options.
+		m3msgAddr = cfg.M3Msg.Server.ListenAddress
+		m3msgInsrumentOpts := instrumentOpts.
+			SetMetricsScope(scope.
+				SubScope("m3msg-server").
+				Tagged(map[string]string{"server": "m3msg"}))
+		m3msgServerOpts, err = cfg.M3Msg.NewServerOptions(m3msgInsrumentOpts)
+		if err != nil {
+			logger.Fatal("could not create m3msg server options", zap.Error(err))
+		}
+	}
 
-	// Create the http server options.
-	httpAddr := cfg.HTTP.ListenAddress
-	httpServerOpts := cfg.HTTP.NewServerOptions()
+	if cfg.RawTCP != nil {
+		// Create the raw TCP server options.
+		rawTCPAddr = cfg.RawTCP.ListenAddress
+		rawTCPInstrumentOpts := instrumentOpts.
+			SetMetricsScope(scope.
+				SubScope("rawtcp-server").
+				Tagged(map[string]string{"server": "rawtcp"}))
+		rawTCPServerOpts = cfg.RawTCP.NewServerOptions(rawTCPInstrumentOpts)
+	}
+
+	if cfg.HTTP != nil {
+		// Create the http server options.
+		httpAddr = cfg.HTTP.ListenAddress
+		httpServerOpts = cfg.HTTP.NewServerOptions()
```

Comment:
nit; Do these need any metrics too?

Reply:
Not for now, this is just management APIs.